sycl/test-e2e/AddressSanitizer/bad-free/bad-free-minus1.cpp

```diff
@@ -1,10 +1,10 @@
 // REQUIRES: linux, cpu || (gpu && level_zero)
 // RUN: %{build} %device_asan_flags -O0 -g -o %t1.out
-// RUN: %force_device_asan_rt %{run} not %t1.out 2>&1 | FileCheck --check-prefixes CHECK,CHECK-DEVICE %s
+// RUN: %force_device_asan_rt %{run} not --crash %t1.out 2>&1 | FileCheck --check-prefixes CHECK,CHECK-DEVICE %s
```

Comment:
Just wondering, why do we need `--crash`? Like is this changes related to using pre-built test binaries?

Reply:
Thanks, I've updated the PR description to explain this.

Reply:
Just to clarify, by "non-existent binary", we are referring to the test binary, right? So, if build fails and there is no test binary, doing `not <non-existent test binary>` would cause LIT to hang and that's why we add `--crash`. Is my understanding correct?

Reply:
Not quite, it's two steps and not one. We use `UseExternalShell=True` in run-only E2E mode (for whatever reasons it's needed). That explains the difference between doing full build/run and run-only, and that is the reason for this change.

Reply:
Ahh.. we use external shell... The change makes sense to me now. Thanks!